client/app/src/main/java/com/example/criengine/Objects/Profile.java

```diff
@@ -42,77 +69,174 @@ public Profile(String userID, String email, String username, String phone, Strin
+     */
     public void setFirstname(String firstname) {
         this.firstname = firstname;
     }
 
+    /**
+     * Get the last name of the user.
+     * @return The last name.
+     */
     public String getLastname() {
         return lastname;
     }
 
+    /**
+     * Set a new last name for the user.
+     * @param lastname The new last name.
+     */
     public void setLastname(String lastname) {
         this.lastname = lastname;
     }
 
+    /**
+     * Get the bio for the profile.
+     * @return The bio.
+     */
     public String getBio() {
         return bio;
     }
 
+    /**
+     * Set a new bio for the profile.
+     * @param bio The new bio.
+     */
     public void setBio(String bio) {
         this.bio = bio;
     }
 
+    /**
+     * Get the email for the profile.
+     * @return The email.
+     */
     public String getEmail() {
         return email;
     }
 
+    /**
+     * Set a new email for the account.
+     * @param email The new email.
+     */
     public void setEmail(String email) {
         this.email = email;
     }
 
+    /**
+     * Get the user identifier.
+     * @return The ID.
+     */
     public String getUserID() {
         return userID;
     }
 
+    /**
+     * Set a new user identifier for the profile.
+     * @param userID The new user ID.
+     */
     public void setUserID(String userID) {
         this.userID = userID;
     }
 
+    /**
+     * Get a list of owned books.
+     * @return The list of owned books.
+     */
     public ArrayList<String> getBooksOwned() {
         return booksOwned;
     }
 
+    /**
+     * Set a new list of owned books.
+     * @param booksOwned The list of owned books.
+     */
     public void setBooksOwned(ArrayList<String> booksOwned) {
         this.booksOwned = booksOwned;
     }
 
+    /**
+     * Get a list of books that the user has borrowed or requested.
+     * @return The list of borrowed or requested books.
+     */
     public ArrayList<String> getBooksBorrowedOrRequested() {
         return booksBorrowedOrRequested;
     }
 
+    /**
+     * Set a list of borrowed or requested books.
+     * @param booksBorrowedOrRequested The list of borrowed or requested books.
+     */
     public void setBooksBorrowedOrRequested(ArrayList<String> booksBorrowedOrRequested) {
         this.booksBorrowedOrRequested = booksBorrowedOrRequested;
     }
+
+    /**
+     * Add a notification for the user.
+     * @param notif The notification to be added.
+     */
+    public void addNotification(Notification notif) {
+        notifications.add(notif);
+    }
+
+    /**
+     * Remove a notification from the user.
+     * @param notif The notification to be removed.
+     */
+    public void removeNotification(Notification notif) {
+        notifications.remove(notif);
+    }
+
+    public ArrayList<Notification> getNotifications() {
+        return notifications;
+    }
+
+    public void setNotifications(ArrayList<Notification> newList) {
```

Comment:
Missed some java doc on these last two methods :p But thanks for writing them!

Reply:
oops!